### ISSUE TITLE
[Ona Support Request]: Adding a new ID field 

### ISSUE THREAD

**Issue by mergygayatri** [OpenSRP/opensrp-client-anc#940]
### Affected App or Server Version

v1.6.18 App

### What kind of support do you need?

We need to add a new field for additional ID numbers in addition to the NIK

### What is the acceptance criteria for your support request?

Additional ID field

**Comment by Naima-Bashir** [comment]
This is noted. It will be picked up once the manual encounter work is complete

**Comment by Naima-Bashir** [comment]
Hi @mergygayatri Kindly answer the following questions in order for us to proceed with this ticket:
1. What is the translation for the new ID field?
2. Should the new ID field show up once a user selects the button that says: Tidak ada nomor KTP (no number)?
3. Should the new ID field be mandatory?
4. What is the length of the new ticket? Will it be 16 like the NIK number?

![image](https://user-images.githubusercontent.com/61143692/233923475-cbe6ce71-4ebd-4acd-8f0d-615c60dc1da5.png)
![image](https://user-images.githubusercontent.com/61143692/233923490-8d2d3f57-04aa-4edd-b049-ef27d87deaf4.png)

cc @hilpitome @SebaMutuku

**Comment by mergygayatri** [comment]
Hi @Naima-Bashir @SebaMutuku @hilpitome 

**Please add 2 new additional fields:**
**_1. Additional ID type (In Bahasa Indonesia: Tipe ID tambahan)_**
- 10 numeric characters
- Enter twice
- Mandatory field

**_2. Additional ID number (In Bahasa Indonesia: Nomor ID tambahan)_**
- 9 numeric characters
- Enter twice
- Mandatory field

**Additional requests:**
1. NIK should be entered twice
2. BPJS number should be entered twice
3. The pregnant woman's mobile number should be entered twice

With regard to the order of the fields, please see the illustration attached. Thank you. 
![Discussion 7052024](https://github.com/opensrp/opensrp-client-anc/assets/119648701/0cf16077-d082-4e42-9b36-79613bd56b75)

**Comment by mergygayatri** [comment]
Hi @Naima-Bashir and @junaidwarsivd, I am sorry I need to resend the additional fields that we need. 
![IMG_0113](https://github.com/opensrp/opensrp-client-anc/assets/119648701/e9b9267d-ec68-4080-9780-6c26d093815b)

**Comment by Naima-Bashir** [comment]
Hi @mergygayatri this is noted. We will remove this feature from the test  apk for now and once the dev work is complete it can be added back. At the moment, we are working on the last minor encounter related bug.

**Comment by junaidwarsivd** [comment]
hi @mergygayatri what does it mean by entering blindly ?  cc @Naima-Bashir

**Comment by Naima-Bashir** [comment]
Hi @mergygayatri could you kindly clarify what blindly means? Thanks

**Comment by Naima-Bashir** [comment]
Hi @mergygayatri Kindly advice on the word blindly

**Comment by SebaMutuku** [comment]
@mergygayatri  do we really need to confirm what the user entered considering that we can show the additional field. The user can as well see what they entered

**Comment by mergygayatri** [comment]
Hi @SebaMutuku. Apologies for the delayed response. We need to enter data two times because we want to make sure that we enter the data correctly but we shouldn’t be able to see the data we’ve inputted in the first field. Also, there is a notification saying that it needs to be reentered if the data in the two fields do not match.

**Comment by Naima-Bashir** [comment]
When one clicks on "Tidak ada nomor KTP" , the Nomor KTP "NIK " should be hidden. 
This is because the Tidak ada nomor KTP stands for "ID unknown/ I don't have ID" cc @junaidwarsivd 
![image](https://github.com/opensrp/opensrp-client-anc/assets/61143692/048a9d81-c532-4399-9498-c7df0189d5a0)

**Comment by Naima-Bashir** [comment]
This has passed QA. Including this section: When one clicks on "Tidak ada nomor KTP" , the Nomor KTP "NIK " should be hidden.

**Comment by Naima-Bashir** [comment]
There's a section: Tidak ada nomor HP that behaves in a similar manner. I think we should also hide the mobile number (Nomor HP) when one says they do not have it.

**Comment by AngelaKabari** [comment]
@mergygayatri has this issue been satisfactorily resolved? If yes, please close this out. Thanks.

**Comment by mergygayatri** [comment]
hi @AngelaKabari could you change the additional ID type from digits into characters? Thank you.

**Comment by AngelaKabari** [comment]
Just to clarify @mergygayatri: 
1. We should change the additional ID type field from digits into characters only and not characters + digits?
2. The field should remain not hidden?

**Comment by mergygayatri** [comment]
Hi Angela. 
In terms of the additional ID type field, please change it into characters and not hidden, while the additional ID number field remains digits and hidden.

**Comment by AngelaKabari** [comment]
Noted. We shall address this once we have completed the more urgent issues around the app log out and team & location assignments.

**Comment by Naima-Bashir** [comment]
Changes:
1. jenis id tambahan (additonal ID type will be changes from numbers to characters)
2. remove the character limit for jenis id tambahan
cc @junaidwarsivd @mergygayatri

**Comment by Naima-Bashir** [comment]
Hi @junaidwarsivd kindly remove the character limit for jenis tambahan (additional ID type)

**Comment by Naima-Bashir** [comment]
This has passed QA. Closing this
